src/main/java/com/r307/arbitrader/service/ticker/StreamingTickerStrategy.java

```diff
@@ -62,21 +66,32 @@ public StreamingTickerStrategy(ErrorCollectorService errorCollectorService) {
     private List<Disposable> subscribeAll(StreamingExchange exchange, List<CurrencyPair> currencyPairs) {
         return currencyPairs
             .stream()
-            .map(pair -> exchange.getStreamingMarketDataService().getTicker(pair).subscribe(
-                ticker -> {
-                    tickers.computeIfAbsent(exchange, e -> new HashMap<>());
-                    tickers.get(exchange).put(pair, ticker);
+            .map(pair -> {
+                // TODO: Temp fix while https://github.com/knowm/XChange/pull/3761 is not merged and released
+                final Observable<Ticker> tickerObservable;
+                final String exchangeName = exchange.getExchangeSpecification().getExchangeName();
+                if (exchangeName.contains("Gemini")) {
```

Comment:
Could we do something like `if (exchange instanceof GeminiStreamingMarketDataService)` instead, so we don't match the non-streaming Gemini or any other exchanges that might pop up in the future with Gemini in their names? This only affects the Gemini streaming exchange specifically, correct?

Reply:
That is actually a good idea, I will push it in a minute.
Yes, so far (from all exchanges I tested) it only happened with Gemini Streaming.